### ISSUE TITLE
css :has() selector doesn't work properly

### ISSUE THREAD

**Issue by cilthepen** [stylus/stylus#2808]
**To reproduce:**

```styl
[class*="parent-"] {
    width: 500px;
}
[class*="parent-"]:has(nav[aria-label*="child"]) {
    width: 100px;
}
[class*="parent-"]:has(nav[aria-label*="child"]):hover {
    width: 250px;
}
```

**Current behavior:**

:has selector doesn't get nav:[aria-label*="child"] as it should be.

**Expected behavior:**

:has selector should get nav[aria-label*="child"].

**Environment information:**

- `stylus` version: 1.5.33
- `nodejs` version: latest

I have been using stylus to customize discord but this :has selector doesn't really select as it should be. When trying on discord app with client mod, my css style does work well and :has selector works well. So, I think  the problem is in Stylus.

**Comment by cilthepen** [comment]
oops, my bad. It doesn't work because i've been using Edge 100 which does not support has selector.
